### PR TITLE
 OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by berylcl**
### Changes

In this pull request, I have added a console application to manage a school library system. The application allows users to perform various actions related to books, people (students and teachers), and rentals.

### Changes in app.rb

- List all books: Displays a list of all available books in the library.
- List all people: Shows a list of all students and teachers registered in the system.
- Create a person: Enables the user to add a new student or teacher to the system.
- Create a book: Allows the user to add a new book to the library.
- Create a rental: Enables the user to create a rental transaction by associating a person with a book.
- List all rentals for a given person id: Displays all rental transactions for a person, identified by their unique ID.

### Changes in main.rb

The file `main.rb` contains the main entry point method called `main`. This method provides a user-friendly interface for interacting with the school library system. When the user runs `main.rb`, they are presented with a list of options to choose from. The options include listing books, people, creating a person, creating a book, creating a rental, listing rentals for a person, and quitting the application.

The `main` method allows users to choose an option and, if required, prompts for additional parameters to execute the selected action. The application provides a way to quit  when the user decides to exit.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    puts "Enter the person's name:"
+    name = gets.chomp
+
+    puts "Enter the person's age:"
+    age = gets.chomp.to_i
+
+    puts 'Is the person a student? (yes/no):'
+    is_student = gets.chomp.downcase == 'yes'
+
+    @people << (is_student ? Student.new(name, age) : Teacher.new(name, age))
```
Comment (t-yanick) [inline app.rb]: - [ ] Great job working on this milestone. While creating a person, your app give two options (Yes/No) Taking No means the person being created is not a student. When I take No, I have an error and the app exits as you can see on the screenshot below. Kindly review and fix this issue.
image

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    puts "#{name} has been added as a #{is_student ? 'student' : 'teacher'}."
```
Comment (t-yanick) [inline app.rb]: - [ ] As per sneak peak video, I think you are supposed to add permission for student and specalization for teacher. This is currently absent in your app. See example where the options for student and teacher are 1 and 2 respectively.
```
 person =
      if person_type == "1"
        print "Has parent permission? [Y/N]: "
        # *If needed, ask for parameters for the option.
        parent_permission = gets.chomp
        parent_permission = parent_permission.downcase == "y"

        Student.new(age, name, parent_permission)
      elsif person_type == "2"
        print "Specialization: "
        # *If needed, ask for parameters for the option.
        specialization = gets.chomp

        Teacher.new(age, specialization, name)
```

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @berylcl , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  main.rb is the entry point of the app
:heavy_check_mark: UI looks similar to sneak peak
:heavy_check_mark: Use of correct gitflow

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    puts 'Is the person a student? (yes/no):'
+    is_student = gets.chomp.downcase
+
+    if is_student == 'yes'
+      puts 'Does the student have parent permission? (yes/no):'
+      parent_permission = gets.chomp.downcase == 'yes'
+      @people << Student.new(name, age, parent_permission: parent_permission)
+      puts "#{name} has been added as a student."
+    end
```
Comment (danmainah) [inline app.rb]: - [ ] @berylcl Great work implementing the change requested however am still not able to create a student I get the error below when I choose the parent permission (yes or no), it seems your student class only accepts 2 parameters on the initialize method.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @berylcl , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️  Implemented the change requested to ensure all menu options in person creation are available.

 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+    puts "Enter the person's age:"
+    age = gets.chomp.to_i
+
+    puts 'Is the person a student? (yes/no):'
+    is_student = gets.chomp.downcase
+
+    if is_student == 'yes'
+      puts 'Does the student have parent permission? (yes/no):'
+      parent_permission = gets.chomp.downcase == 'yes'
```
Comment (MaganaAsiati) [inline app.rb:38]: - You did a great job by implementing the required changes from the previous reviewer however the app is not working as expected, please make sure that you are able to create a new person as you see in the picture below it brings an error when I'm trying to create a new person. please fix that :+1: 

image

**Review by MaganaAsiati — CHANGES_REQUESTED** [review]
# STATUS: Required Changes ♻️♻️♻️ 

Hi @berylcl 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

AlonzoLeroneThisCloseGIF

## Highlights
✅ No linter error
✅ README is professional

## Required Changes ♻️

- Please make sure all the options are working as expected to meet the requirements of the project. Check the school library console app behaves here  [example video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) .
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @berylcl , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ All linters checks are ok
✅ The correct workflow is used. 
✅ Very descriptive Pull Request(PR)
✅ Your work is well documented.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**